File: inventario_frontend/src/models/productEntry.ts
import { paymentTypes } from "@/enums/paymentTypes";
import { ProductEntryLine } from "./productEntryLine";
import { Vendor } from "./vendor";

export class ProductEntryForCreation {
  public date: Date | undefined = undefined;
  public vendorId?: string;
  public cost?: number;
  public paymentType?: paymentTypes;
  public observations?: string;
  public isEntry = true;
  public productEntryLines: ProductEntryLine[] = [];
}

export class ProductEntry extends ProductEntryForCreation {
  public id!: string;
  public vendor?: Vendor;
}

File: Inventario/DataAccess/Models/ProductEntry.cs
using iText.Kernel.XMP.Properties;
using System;
using System.Collections.Generic;
using System.Text;
using Util.Enums;

namespace Data
{
    public class ProductEntry : Entity
    {
        public DateTime Date { get; set; }
        public bool IsEntry { get; set; }
        public Guid? VendorId { get; set; }
        public Vendor? Vendor { get; set; }
        public decimal? Cost { get; set; }
        public ePaymentTypes? PaymentType { get; set; }
        public string Observations { get; set; }
        public List<ProductEntryLine> ProductEntryLines { get; set; }
        public ProductEntry() : base()
        {
            ProductEntryLines = new List<ProductEntryLine>();
        }
    }
}
